rest/route/agent.go

```diff
@@ -368,7 +368,30 @@ func (h *getExpansionsAndVarsHandler) Run(ctx context.Context) gimlet.Responder
 		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "getting GitHub OAuth token"))
 	}
 
-	e, err := model.PopulateExpansions(t, foundHost, oauthToken)
+	v, err := model.VersionFindOne(model.VersionById(t.Version))
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding version '%s'", t.Version))
+	}
+	if v == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("version '%s' not found", t.Version),
+		})
+	}
+
+	appToken, err := h.settings.CreateInstallationToken(ctx, v.Owner, v.Repo, nil)
```

Comment:
do you think it would be worth it to make an extra db call to get the projectref? 

i did notice that it wasnt being set or used so i had to make some changes to set them in version code

Reply:
A find one for project ref should be almost instantaneous and not expensive for the DB.

I wouldn't change the logic to set them in the version just for this one piece of code. Deprecating them and using the project ref seems preferrable.

Reply:
updated